packages/vercel/src/index.ts

```diff
@@ -204,6 +206,29 @@ export default function vercelAdapter({
 					);
 				}
 
+				staticDir = new URL('./.vercel/output/static', config.root);
+				updateConfig({
+					build: {
+						format: 'directory',
+						redirects: false,
+					},
+					vite: {
+						ssr: {
+							external: ['@vercel/nft'],
+						},
+					},
+					...getAstroImageConfig(
+						imageService,
+						imagesConfig,
+						command,
+						devImageService,
+						config.image
+					),
+				});
+			},
+			'astro:config:done': ({ setAdapter, config, logger }) => {
+				buildOutput = config.output;
```

Comment:
I think you can get the final `buildOutput` directly from the `astro:config:done` hook instead. [It returns `buildOutput`](https://github.com/withastro/astro/blob/d63d87dcaea85e84acb26b70cbd650d6e1eea6df/packages/astro/src/types/public/integrations.ts#L187)